File: aat-android/src/main/java/ch/bailu/aat/util/AndroidLoggerFactory.java
package ch.bailu.aat.util;

import android.util.Log;

import ch.bailu.aat_lib.logger.Logger;
import ch.bailu.aat_lib.logger.LoggerFactory;

public class AndroidLoggerFactory implements LoggerFactory {
    @Override
    public Logger warn() {
        return Log::w;
    }

    @Override
    public Logger info() {
        return Log::i;
    }

    @Override
    public Logger debug() {
        return Log::d;
    }

    @Override
    public Logger error() {
        return Log::e;
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/util/AndroidLoggerFactory.kt
package ch.bailu.aat.util

import android.util.Log
import ch.bailu.aat_lib.logger.Logger
import ch.bailu.aat_lib.logger.LoggerFactory

class AndroidLoggerFactory : LoggerFactory {
    override fun warn(): Logger {
        return Logger { tag: String, msg: String -> Log.w(tag, msg) }
    }

    override fun info(): Logger {
        return Logger { tag: String, msg: String -> Log.i(tag, msg) }
    }

    override fun debug(): Logger {
        return Logger { tag: String, msg: String -> Log.d(tag, msg) }
    }

    override fun error(): Logger {
        return Logger { tag: String, msg: String -> Log.e(tag, msg) }
    }
}
